### DISCUSSION TITLE
none allows XAuthInitPSK authentication using Main Mode

### DISCUSSION THREAD

**Discussion by gityms123** [strongswan/strongswan#1081] [Q&A]
I set aggressive=yes in ipsec.conf and i_dont_care_about_security_and_use_aggressive_mode_psk=yes in strongswan.conf, the connection still has such an error

**Comment by tobiasbrunner** [comment]
If your peer initiates a Main Mode SA, as the log message in the title indicates, then `aggressive=yes` is clearly not the right setting.

**Comment by Raymondzfy** [comment]
I ran into the same issue that installed strongswan-5.9.6-1.el8.x86_64 on centos Steam 8 with below config in /etc/strongswan/swanctl/conf.d/cisco_ipsec.conf

cat /etc/strongswan/swanctl/conf.d/cisco_ipsec.conf

  cisco_ipsec {
    pools = nodepool
    fragmentation = yes
    rekey_time = 0
    version = 1
    rekey_time = 500h
    #aggressive = yes
    local_addrs = 0.0.0.0/0,::/0
    
    local {
        auth = psk
        certs = /etc/strongswan/ipsec.d/certs/test.crt
    }
    children {
      nodechild {
        local_ts = 0.0.0.0/0
        dpd_action = clear
        life_time = 200s
        esp_proposals = 3des-sha1
        start_action = none
      }
    }
    remote_addrs = 0.0.0.0/0,::/0
    remote {
      auth = psk
    }
    remote_cisco_ipsec {
      auth = eap-mschapv2
    }
  }


When I connect from Mac client with Cisco IPSec connect, and I got below same errors:
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received packet: from 98.114.241.25[500] to 45.56.107.45[500] (848 bytes)
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: parsed ID_PROT request 0 [ SA V V V V V V V V V V V V V V ]
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received NAT-T (RFC 3947) vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received draft-ietf-ipsec-nat-t-ike vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received draft-ietf-ipsec-nat-t-ike-08 vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received draft-ietf-ipsec-nat-t-ike-07 vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received draft-ietf-ipsec-nat-t-ike-06 vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received draft-ietf-ipsec-nat-t-ike-05 vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received draft-ietf-ipsec-nat-t-ike-04 vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received draft-ietf-ipsec-nat-t-ike-03 vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received draft-ietf-ipsec-nat-t-ike-02 vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received draft-ietf-ipsec-nat-t-ike-02\n vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received XAuth vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received Cisco Unity vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received FRAGMENTATION vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received DPD vendor ID
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: 98.114.241.25 is initiating a Main Mode IKE_SA
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: generating ID_PROT response 0 [ SA V V V V ]
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: sending packet: from 45.56.107.45[500] to 98.114.241.25[500] (160 bytes)
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received packet: from 98.114.241.25[500] to 45.56.107.45[500] (380 bytes)
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: parsed ID_PROT request 0 [ KE No NAT-D NAT-D ]
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: remote host is behind NAT
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: generating ID_PROT response 0 [ KE No NAT-D NAT-D ]
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: sending packet: from 45.56.107.45[500] to 98.114.241.25[500] (396 bytes)
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: received packet: from 98.114.241.25[4500] to 45.56.107.45[4500] (108 bytes)
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: parsed ID_PROT request 0 [ ID HASH N(INITIAL_CONTACT) ]
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: looking for XAuthInitPSK peer configs matching 45.56.107.45...98.114.241.25[192.168.1.167]
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: found 1 matching config, but none allows XAuthInitPSK authentication using Main Mode
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: generating INFORMATIONAL_V1 request 424490140 [ HASH N(AUTH_FAILED) ]
Oct 29 13:43:15  testnode.example.org charon-systemd[196930]: sending packet: from 45.56.107.45[4500] to 98.114.241.25[4500] (108 bytes)



Any ideas what were the issues?

Thanks

**Reply by Thermi** [reply]
```
remote_cisco_ipsec {
  auth = eap-mschapv2
}
```

That's not the right one.

Supposedly you are acting as  responder here. So you want 

```
remote-cisco-ipsec {
auth = xauth-generic
}
```
The dash following `remote` is important, as is the order of the sections.
eap-mschapv2 will not work because eap is IKEv2 only.

**Comment by Raymondzfy** [comment]
Thanks so much, I fixed the issues after I applied the change!
